sw/device/tests/sim_dv/chip_sw_lc_ctrl_scrap_test.c

```diff
@@ -68,43 +48,24 @@ bool execute_lc_ctrl_scrap_test(bool use_ext_clk) {
   mmio_region_t lc_reg = mmio_region_from_addr(TOP_EARLGREY_LC_CTRL_BASE_ADDR);
   CHECK_DIF_OK(dif_lc_ctrl_init(lc_reg, &lc));
 
-  LOG_INFO("Read and check LC state.");
-  dif_lc_ctrl_state_t curr_state;
-  CHECK_DIF_OK(dif_lc_ctrl_get_state(&lc, &curr_state));
-
-  // The OTP preload image hardcodes the initial LC state transition count to 8.
-  // With each iteration of test, there are two LC_CTRL state transitions.
-  // And the first LC_CTRL state transition is done via external JTAG interface.
-  // `kTestIterationCount` starts with 1 in SystemVerilog.
-  const uint8_t kLcStateTransitionCount = 8 + 1 + (kTestIterationCount - 1) * 2;
-
-  if (curr_state == kDifLcCtrlStateTestUnlocked2) {
-    // LC TestUnlocked2 is the intial test state for this sequence.
-    // The sequence will check if lc_count matches the preload value.
-    lc_ctrl_testutils_check_transition_count(&lc, kLcStateTransitionCount);
+  // acquire the mutex and perform a transition to SCRAP
```

Comment:
are you relying on the mutex here to decide whether the SW side should move forward and perform the transition? I don't see any other synchronization mechanism...

It may be better to use an explicit symbol SW symbol override from the SV side to communicate to SW that the transition is done either in SW or SV, see here for example:
https://github.com/lowRISC/opentitan/blob/407d32b4264b02d5821b21be9d178265be9e06d5/hw/top_earlgrey/dv/env/seq_lib/chip_sw_data_integrity_vseq.sv#L63

https://github.com/lowRISC/opentitan/blob/407d32b4264b02d5821b21be9d178265be9e06d5/sw/device/tests/sim_dv/data_integrity_escalation_reset_test.c#L537-L541

Reply:
hi @msfschaffner, I indeed use the mutex to move forward or block the transition through SW. 

Main idea was that if the transition is being performed via JTAG it wouldn't be possible to perform the transition through SW anyway because either the mutex is already locked in relevant states by JTAG or it's not possible to execute SW since we're in a state that doesn't permit that.

However, I see the loophole in source states that allow SW execution, so I'll add a synchronization mechanism via SV.

Reply:
Thanks for making the change - this looks good!